soundboard/src/main/java/editor/model/EditableCatalogue.java

```diff
@@ -68,4 +91,22 @@ public void clearChanges() {
         markedForRemoval.clear();
         recentlyAdded.clear();
     }
+
+    public Catalogue getCatalogue() {
+        return catalogue;
+    }
+
+    public List<Group> getRecentlyAdded() {
+        return recentlyAdded;
+    }
+
+    public boolean hasGroup(String groupName) {
+        if (recentlyAdded.stream().anyMatch(group-> group.getName().equals(groupName))){
+            return true;
+        }
+        if (editedGroups.stream().anyMatch(group-> group.getUpdatedName().equals(groupName))){
+            return true;
+        }
+        return catalogue.stream().anyMatch(group -> group.getName().equals(groupName));
```

Comment:
There might be a better way to do this, but I don't see one right now. Best I've got is to create  a distinct stream combining the String names from each of these three collections into one then calling equals.